eng/tools/suppressions/src/index.ts

```diff
@@ -0,0 +1,141 @@
+import { access, constants, readFile } from "fs/promises";
+import { minimatch } from "minimatch";
+import { dirname, join, resolve } from "path";
+import { exit } from "process";
+import { parse as yamlParse } from "yaml";
+import { z } from "zod";
+import { fromError } from "zod-validation-error";
+
+interface Suppression {
+  tool: string;
+  path: string;
+  reason: string;
+}
+
+const suppressionSchema = z.array(
+  z.object({
+    tool: z.string(),
+    path: z.string(),
+    reason: z.string(),
+  }),
+);
+
+export async function main() {
+  const args = process.argv.slice(2);
+
+  if (args.length === 2) {
+    const tool = args[0];
+    const path = args[1];
+    const suppressions = await getSuppressions(tool, path);
+    console.log(JSON.stringify(suppressions));
+    exit(0);
+  } else {
+    console.error(getUsage());
+    exit(1);
+  }
+}
+
+function getUsage(): string {
+  return (
+    "  Usage: npx get-suppressions <tool-name> <path-to-file>\n" +
+    "Returns: JSON array of suppressions, with specified tool name, applying to file (may be empty)\n" +
+    "\n" +
+    "Example: npx get-suppressions TypeSpecRequirement specification/foo/data-plane/Foo/stable/2023-01-01/Foo.json\n" +
+    'Returns: [{"tool":"TypeSpecRequirement","path":"**/*.json","reason":"foo"}]\n'
+  );
+}
+
+// tool: Name of tool
+// example: TypeSpecRequirement
+//
+// path: Path to file under analysis
+// example: specification/foo/data-plane/Foo/stable/2024-01-01/foo.json
+//
+// returns: Array of suppressions matching tool and path (may be empty)
+// example: [{"tool":"TypeSpecRequirement","path":"**/*.json","reason":"foo"}]
+export async function getSuppressions(tool: string, path: string): Promise<Suppression[]> {
+  path = resolve(path);
+
+  // If path doesn't exist, throw instead of returning "[]" to prevent confusion
+  await access(path, constants.R_OK);
+
+  let suppressionsFile = await findSuppressionsYaml(path);
```

Comment:
```suggestion
  let suppressionsFile: string | undefined = await findSuppressionsYaml(path);
```

If you are not particularly opposed to this :)

Reply:
Just here or everywhere?

Reply:
@mikeharder everywhere. Sorry, I will be less ambiguous next time. I avoid writing out types explicitly only if they are super-long and for some reason I don't want to introduce corresponding `type` or `interface`.

Reply:
https://github.com/Azure/azure-rest-api-specs/issues/28975